### PR TITLE
Feature/issue 44/enhancements bring application appplication context

### LINKED ISSUE #44: BringApplication and ApplicationContext class enhancements
1. Simplify BringApplication#run
Remove createApplication method
then
 public void run() {
    [log.info](http://log.info/)("Starting Bring application...");
    applicationContext = new AnnotationConfigApplicationContext();
 }
2. BringApplication.run() should return ApplicationContext
3. Pass package to scan into BringApplication and ApplicationContext
4. Make BringApplication in a builder pattern_(optional)_
5. Add log about that fact that application has been started successfully

### PR DESCRIPTION
**Pull request by marynamelnichuk**
## Checklist before requesting a review
- [X] There are no MR remarks
- [X] Clean and concise code that is easy to read and understand
- [X] Code covered with tests and they are green
- [X] Code API has informative JavaDoc on it
- [X] Rich logging with detailed messages and correct use of log levels

### CODE REVIEW

File: src/test/java/com/bringframework/integration/IntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.bringframework.integration.service.IMessageService;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class IntegrationTest {
+
+  @Test
+  void test() {
```
Comment (olehhaliak) [inline src/test/java/com/bringframework/integration/IntegrationTest.java]: Could you please give a more meaningful name for this test

File: src/test/java/com/bringframework/integration/IntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.bringframework.context.ApplicationContext;
+import com.bringframework.integration.service.ChatService;
+import com.bringframework.integration.service.GreetingService;
+import com.bringframework.integration.service.IMessageService;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class IntegrationTest {
```
Comment (mvoronov) [inline src/test/java/com/bringframework/integration/IntegrationTest.java]: Integration tests by convention have postfix `IT`. 
Lets rename this test class to  `BringApplicationIT`

**Review by vitshch — APPROVED** [review]

File: src/test/java/com/bringframework/reader/DefaultBeanDefinitionReaderTest.java [hunk 3]
```diff
@@ -32,7 +32,7 @@ class DefaultBeanDefinitionReaderTest {
 
   @Test
   void shouldRegisterBeans() {
-    underTest.registerBeans("com.bringframework");
+    underTest.registerBeans("com.bringframework.reader");
```
Comment (mvoronov) [inline src/test/java/com/bringframework/reader/DefaultBeanDefinitionReaderTest.java]: Was it change required in this PR? If not, then please revert it
btw, I saw that @olehhaliak did renaming of BeanReader to BeanScanner, so potentially here you would have unnecessary conflict

**Review by mvoronov — CHANGES_REQUESTED** [review]

**Comment by financeasist** [comment]
please synchronize your changes with latest changes from dev

File: src/main/java/com/bringframework/context/AnnotationConfigApplicationContext.java [hunk 4]
```diff
@@ -11,23 +14,23 @@
...
+  private final Map<String, Object> beans;
 
-  public AnnotationConfigApplicationContext() {
+  public AnnotationConfigApplicationContext(String packageName) {
+    validatePackageName(packageName);
     log.trace("Application context is collecting...");
     BeanDefinitionRegistry beanDefinitionRegistry = new DefaultBeanDefinitionRegistry();
-    BeanDefinitionReader beanDefinitionReader = new DefaultBeanDefinitionReader(
-        beanDefinitionRegistry);
+    new DefaultBeanDefinitionReader(beanDefinitionRegistry).registerBeans(packageName);
```
Comment (financeasist) [inline src/main/java/com/bringframework/context/AnnotationConfigApplicationContext.java]: it is already `BeanDefinitionScanner`, please update with latest changes from dev and resolve the conflicts

**Review by financeasist — CHANGES_REQUESTED** [review]

File: src/test/java/com/bringframework/reader/DefaultBeanDefinitionReaderTest.java (hunk 3 above)
Comment (marynamelnichuk) [reply src/test/java/com/bringframework/reader/DefaultBeanDefinitionReaderTest.java]: It's required, cause tests were failing

**Review by mvoronov — APPROVED** [review]

**Review by MykolaHarasym — APPROVED** [review]

**Review by financeasist — APPROVED** [review]

**Pull request merged**